packages/amplify-graphql-relational-transformer/src/graphql-has-many-transformer.ts

```diff
@@ -158,6 +159,14 @@ export class HasManyTransformer extends TransformerPluginBase {
 
 const validate = (config: HasManyDirectiveConfiguration, ctx: TransformerContextProvider): void => {
   const { field } = config;
+  if (!ctx.transformParameters.allowGen1Patterns) {
+    if (field.type.kind === Kind.NON_NULL_TYPE) {
+      throw new InvalidDirectiveError(`@${HasManyDirective.name} cannot be used on required fields.`);
+    }
+    if (config.fields) {
+      throw new InvalidDirectiveError(`fields argument on @${HasManyDirective.name} is deprecated. Use references instead.`);
```

Comment:
Here and throughout all error messages: It's not deprecated, it's outright forbidden in Gen2 projects.